training/src/test/java/wolox/training/controllers/UsersControllerTest.java

```diff
@@ -0,0 +1,163 @@
+        oneTestBook.setPages(30);
+        oneTestBook.setPublisher(" publisher 1234");
+        oneTestBook.setGenre("ficcion");
+        oneTestBook.setAuthor("J.K rowling");
+        oneTestBook.setIsbn("isgn 12");
+        oneTestBook.setImage("one image");
+
+        // a book test withputUser
+        oneTestBookWithoutUser = new Book();
+        oneTestBookWithoutUser.setId(2L);
+        oneTestBookWithoutUser.setYear("1999");
+        oneTestBookWithoutUser.setTitle("Harry potter 2");
+        oneTestBookWithoutUser.setSubtitle("la orden del fenix");
+        oneTestBookWithoutUser.setPages(30);
+        oneTestBookWithoutUser.setPublisher(" publisher 22222");
+        oneTestBookWithoutUser.setGenre("ficcion");
+        oneTestBookWithoutUser.setAuthor("J.K rowling");
+        oneTestBookWithoutUser.setIsbn("isgn 12");
+        oneTestBookWithoutUser.setImage("one image");
+
+        // user with Id
+        oneTestUser = new Users();
+        oneTestUser.setId(1L);
+        oneTestUser.setName("pedro");
+        oneTestUser.setUsername("pedro1234");
+        oneTestUser.setBirthdate(LocalDate.parse("1993-12-06"));
+        oneTestUser.addBook(oneTestBook);
+
+    }
+
+    @Test
+    void whenFindByIdWhichExists_thenUserIsReturned() throws Exception {
+        Mockito.when(mockUserRepository.findById(1L))
+            .thenReturn(java.util.Optional.ofNullable(oneTestUser));
+        String url = ("/api/books/users/1");
+        mvc.perform(get(url)
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.username", is(oneTestUser.getUsername())))
+            .andExpect(jsonPath("$.name", is(oneTestUser.getName())))
+            .andExpect(jsonPath("$.birthdate", is(oneTestUser.getBirthdate().toString())));
+    }
+
+    @Test
+    void whenFindByIdWithOutId_thenThrowException() throws Exception {
+        Mockito.when(mockUserRepository.findById(1L))
+            .thenReturn(java.util.Optional.ofNullable(oneTestUser));
+        String url = ("/api/books/users/0");
+        mvc.perform(get(url)
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isNotFound());
+    }
+
+    @Test
+    void whenCreateUserWithId_thenUserIsReturn() throws Exception {
+        Mockito.when(mockUserRepository.save(Mockito.any()))
+            .thenReturn(oneTestUser);
+
+        String url = ("/api/books/users");
+        mvc.perform(
+            post(url).contentType(MediaType.APPLICATION_JSON)
+                .content(BODY_WITH_USER))
+            .andExpect(status().isCreated())
+            .andExpect(jsonPath("$.username", is(oneTestUser.getUsername())))
+            .andExpect(jsonPath("$.name", is(oneTestUser.getName())))
+            .andExpect(jsonPath("$.birthdate", is(oneTestUser.getBirthdate().toString())));
+    }
+
+    @Test
+    void whenUpdateBookOfUser_thenUserIsReturnWithaNewBook() throws Exception {
```

Comment:
Test the case in which the book is not found. To check this properly, in addition to comparing the status code, you should also check that the status reason matches with "book not found" and not with "user not found".